tests/spec/fs.mkdir.spec.js

```diff
@@ -46,3 +46,51 @@ describe('fs.mkdir', function() {
     });
   });
 });
+
+describe('fs.promises.mkdir', function () {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function () {
+    var fs = util.fs();
+    expect(fs.promises.mkdir).to.be.a('function');
+  });
+
+  it('should return an error if part of the parent path does not exist', function (done) {
+    var fs = util.fs();
+
+    fs.promises.mkdir('/tmp/mydir', function (error) {
+      expect(error).to.exist;
+      expect(error.code).to.equal('ENOENT');
+    }).finally(done);
+  });
+
+  it('should return an error if the path already exists', function (done) {
+    var fs = util.fs();
+
+    fs.promises.mkdir('/', function (error) {
```

Comment:
Same issue here.  You want to return your Promise, and use then/catch vs. a callback.